server/ServerFSM.cpp

```diff
@@ -1529,6 +1529,17 @@ sc::result WaitingForMPGameJoiners::react(const CheckStartConditions& u) {
     return discard_event();
 }
 
+sc::result WaitingForMPGameJoiners::react(const Hostless& msg) {
```

Comment:
when does this occur such that handling it is needed?

Reply:
It possible when AI process fails at start.

Reply:
Can you outline the sequence of events here? Like the server is in the MP lobby, the players setup and start a game with AIs, those AIs fail to start up, the server notices, and somewhere posts itself a `Hostless` event to revert to the MP lobby?

Also, should `Idle` have two `Hostless` reactions declared here:

https://github.com/freeorion/freeorion/blob/master/server/ServerFSM.h#L147

Reply:
`Hostless` event is generated here https://github.com/freeorion/freeorion/blob/master/server/ServerFSM.cpp#L274 instead `ShutdownServer`.
I'll delete doubled `Hostless` reacton. It's a mistake.

Reply:
That's where it is generated in the code, but in what situation / series of events would that occur?

Reply:
Server goes from `MPLobby` to `WaitingForMPGameJoiners`, launches AIs. They connects but quickly fails so the server gets disconnection. They server decides to shutdown but for hostless mode it replaced by `Hostless` event.